Makefile

```diff
@@ -6,10 +6,50 @@
 #
 # SPDX-License-Identifier: ISC
 
+CXXFLAGS += -Wno-attributes -O3
+LDFLAGS += -lantlr4-runtime
+
+.PHONY: all
+all: parse_fasm fasm/parse_fasm.so
+
 PYTHON_FORMAT ?= yapf
 format:
 	$(IN_ENV) find . -name \*.py $(FORMAT_EXCLUDE) -print0 | xargs -0 -P $$(nproc) yapf -p -i
 
 check-license:
 	@./.github/check_license.sh
 	@./.github/check_python_scripts.sh
+
+
+parse_fasm: cpp/ParseFasm.o cpp/FasmParser.o cpp/FasmLexer.o
+	$(CXX) $(CXXFLAGS) $(LDFLAGS) $^ -o $@
+
+fasm/parse_fasm.so: cpp/ParseFasm.o cpp/FasmParser.o cpp/FasmLexer.o
+	$(CXX) $(CXXFLAGS) $(LDFLAGS) $^ -fPIC -shared -o $@
+
+cpp/ParseFasm.o: cpp/FasmLexer.h cpp/FasmParser.h
+
```

Comment:
I have never seen a Makefile formatted this way before.

Can you use the more traditional;
```Makefile

HEADERS = cpp/FasmParser.h cpp/FasmParserListener.h ....

$(HEADERS) $(SRC): fasm/FasmParser.g4 cpp/FasmLexer.tokens
```

Reply:
Because we are going to be targetting Windows, I've suggested the Makefile use CMake instead.  That would resolve this particular comment.

Reply:
I've fixed this. I'm working on a CMakeList.txt and will add it if it doesn't take long.

Reply:
I've switched the build to CMake.